crates/optimism/payload/src/builder.rs

```diff
@@ -870,8 +879,13 @@ where
         let mut evm = self.evm_config.evm_with_env(&mut *db, env);
 
         while let Some(tx) = best_txs.next(()) {
-            // ensure we still have capacity for this transaction
-            if info.cumulative_gas_used + tx.gas_limit() > block_gas_limit {
+            let tx_exceeds_da_size = tx_da_limit
+                .is_some_and(|da_limit| tx.compressed_size() > da_limit);
+            let tx_exceeds_block_size = info.cumulative_gas_used + tx.gas_limit() > block_gas_limit;
+            let tx_exceeds_block_da_size = block_da_limit
+                .is_some_and(|da_limit| info.cumulative_da_bytes_used + tx.compressed_size() > da_limit);
+
+            if tx_exceeds_block_size || tx_exceeds_block_da_size || tx_exceeds_da_size {
```

Comment:
I'm not sure if this will be optimized by the compiler, but I think this is fine for now.

we could make this nicer by moving this into a new function of executioninfo

Reply:
moved it to execution info and fixed the short circuiting, but lmk if you had something else in mind and I can change it